crates/ark/src/lsp/statement_range.rs

```diff
@@ -240,6 +186,49 @@ fn find_roxygen_comment_at_point<'tree>(
     return Some((node, code));
 }
 
+/// Assuming `node` is the first node on a line, `expand_across_semicolons()`
+/// checks to see if there are any other non-comment nodes after `node` that
+/// share its line number. If there are, that means the nodes are separated by
+/// a `;`, and that we should expand the range to also include the node after
+/// the `;`.
+fn expand_range_across_semicolons(mut node: Node) -> tree_sitter::Range {
+    let start_byte = node.start_byte();
+    let start_point = node.start_position();
+
+    let mut end_byte = node.end_byte();
+    let mut end_point = node.end_position();
+
+    // We know `node` is at the start of a line, but it's possible the node
+    // ends with a `;` and needs to be extended
+    while let Some(next) = node.next_sibling() {
+        let next_start_point = next.start_position();
+
+        if end_point.row != next_start_point.row {
+            // Next sibling is on a different row, we are safe
+            break;
+        }
+        if next.is_comment() {
+            // Next sibling is a trailing comment, we are safe
+            break;
+        }
+
+        // Next sibling is on the same line as us, must be separated
+        // by a semicolon. Extend end of range to include next sibling.
+        end_byte = next.end_byte();
+        end_point = next.end_position();
+
+        // Update ending `node` and recheck (i.e. `1; 2; 3`)
+        node = next;
+    }
+
+    tree_sitter::Range {
+        start_byte,
+        end_byte,
+        start_point,
+        end_point,
+    }
+}
```

Comment:
Here is the magic we use to "expand" the range of the `node` we typically get back to also include support for trailing `;`

It's basically: "if there is a sibling node on the same line as us, expand the range to include that sibling node"